<launch>
  <node name="bluesea2_1" pkg="bluesea2" type="bluesea2_node" output="screen" >
    #ROS#
    <!-- <param name="topic" value="scan_akira1"/> -->
    <param name="topic" value="scan_raw"/>
    <param name="frame_id" value="laser" />
    <param name="ax_topic" value="/ax_laser_scan"/>
    <param name="ax_frame_id" value="horizontal_laser_link"/>
      #DATA#
    <param name="logPath" value="/tmp/log"/>
    <param name="raw_bytes" value="3"/>
    <param name="min_dist" value="0.1"/>
    <param name="max_dist" value="50.0"/>
    <param name="zero_shift" value="0"/>
    <param name="from_zero" value="false"/>
    <param name="output_scan" value="true" />
    <param name="output_ax_scan" value="true"/>
    <param name="output_cloud" value="false"/>
    <param name="output_360" value="true" />
    <param name="with_checksum" value="true"/>
    <param name="inverted" value="true"/>
    <param name="reversed" value="false"/>
    <param name="hard_resample" value="false"/>
    <param name="soft_resample" value="false"/>
    <param name="with_angle_filter" value="false"/>
    <param name="min_angle" value="-3.1415926"/>
    <param name="max_angle" value="3.1415926"/> 
    <rosparam param="mask1" >[1.57,3.14]</rosparam>
    <rosparam param="mask2" >[-1,0]</rosparam> 
    <param name="error_circle" value="3"/>
    <param name="error_scale" value="0.9"/>
    #CONNECT#
    <!-- <param name="type" value="udp" />
    <param name="lidar_port" value="6543" />
    <param name="local_port" value="6669" />
    <param name="lidar_ip" value="192.168.0.99"/>
    <param name="group_listener" value="false" />
    <param name="group_ip" value="224.0.0.11" /> -->
    <param name="type" value="uart"/>
    <param name="port" value="/dev/ttyUSB0"/>
    <param name="baud_rate" value="768000"/> #rate_list[ 768000, 1000000 ]#
    #GET#
    <param name="uuid" value="-1" />
    #SET#
    <param name="rpm" value="-1"/>
    <param name="resample_res" value="-1"/>
    <param name="with_smooth" value="-1" />
    <param name="with_deshadow" value="-1" />
    <param name="alarm_msg" value="-1"/>
    <param name="direction" value="-1"/>
  </node>

  <node name="bluesea2_2" pkg="bluesea2" type="bluesea2_node" output="screen" >
    #ROS#
    <param name="topic" value="scan_raw_up"/>
    <param name="frame_id" value="laser_up" />
      #DATA#
    <param name="logPath" value="/tmp/log"/>
    <param name="raw_bytes" value="3"/>
    <param name="min_dist" value="0.1"/>
    <param name="max_dist" value="50.0"/>
    <param name="zero_shift" value="0"/>
    <param name="from_zero" value="false"/>
    <param name="output_scan" value="true" />
    <param name="output_cloud" value="false"/>
    <param name="output_360" value="true" />
    <param name="with_checksum" value="true"/>
    <param name="inverted" value="true"/>
    <param name="reversed" value="false"/>
    <param name="hard_resample" value="false"/>
    <param name="soft_resample" value="false"/>
    <param name="with_angle_filter" value="false"/>
    <param name="min_angle" value="-3.1415926"/>
    <param name="max_angle" value="3.1415926"/> 
    <rosparam param="mask1" >[1.57,3.14]</rosparam>
    <rosparam param="mask2" >[-1,0]</rosparam> 
    <param name="error_circle" value="3"/>
    <param name="error_scale" value="0.9"/>
    #CONNECT#
    <!-- <param name="type" value="udp" />
    <param name="lidar_port" value="6542" />
    <param name="local_port" value="6668" />
    <param name="lidar_ip" value="192.168.0.98"/>
    <param name="group_listener" value="false" />
    <param name="group_ip" value="224.0.0.11" /> -->
    <param name="type" value="uart"/>
    <param name="port" value="/dev/ttyUSB1"/>
    <param name="baud_rate" value="230400"/> #rate_list[ 768000, 1000000 ]#
    #GET#
    <param name="uuid" value="-1" />
    #SET#
    <param name="rpm" value="-1"/>
    <param name="resample_res" value="-1"/>
    <param name="with_smooth" value="-1" />
    <param name="with_deshadow" value="-1" />
    <param name="alarm_msg" value="-1"/>
    <param name="direction" value="-1"/>
  </node>

  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base2laserup" args="-0.245 0 0.835 3.1415926 0 0 base_link laser_up"/> -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="base2laserup" args="-0.245 0 0.835 0 0 0 base_link laser_up"/>
</launch>
